src/DurableTask.Core/Tracing/TraceHelper.cs

```diff
@@ -181,13 +173,21 @@ static TraceHelper()
             // Adding additional tags for a SubOrchestrationInstanceFailedEvent
             if (failedEvent != null)
             {
-                activity.SetTag("otel.status_code", "ERROR");
-
-                FailureDetails? failureDetails = failedEvent.FailureDetails;
-                if (failureDetails != null)
+                string? statusDescription = "";
```

Comment:
You can remove the `?` here since `statusDescription` is never null.
```suggestion
                string statusDescription = "";
```